clients/python/src/model_registry/_client.py

```diff
@@ -164,6 +166,73 @@ async def _register_model_artifact(
         return await self._api.upsert_model_version_artifact(
             ModelArtifact(name=name, uri=uri, **kwargs), mv.id
         )
+    
+    def _upload_to_s3(
+        self,
+        artifact_local_path: str,
+        destination: str,
+        region_name: str | None = None,
+    ) -> str:
+        """
+        Uploads a file to an S3 bucket.
+        """
+        pass
+
+    def _upload_to_oci(
+        self,
+        artifact_local_path: str,
```

Comment:
One thing to consider here, is how to pass/customize the "base image" of the modelcar being built. Do we want that to be an explicit parameter, or a default from a config which could be overridden if needed ?